src/nimble.nim

```diff
@@ -214,13 +214,20 @@ proc processDeps(pkginfo: PackageInfo, options: Options): seq[PackageInfo] =
     addRevDep(options.nimbleData, i, pkginfo)
 
 proc buildFromDir(pkgInfo: PackageInfo, paths: seq[string],
-                  args: var seq[string]) =
+                  args: var seq[string], options: Options) =
+  doAssert(options.action.typ == actionBuild)
```

Comment:
I'm surprised this doesn't break when calling ``nimble install``. Either the test coverage is bad or I'm missing something. But why did you add these asserts?

Reply:
Sanity check. I've checked all the entry points and the assert doesn't fire.